optaplanner-examples/src/main/java/org/optaplanner/examples/vehiclerouting/domain/CustomerOrVehicle.java

```diff
@@ -0,0 +1,7 @@
+package org.optaplanner.examples.vehiclerouting.domain;
+
+import org.optaplanner.examples.vehiclerouting.domain.location.Location;
+
+public interface CustomerOrVehicle {
```

Comment:
This is starting to smell of the old chained model.
Don't break my brain thinking that something is a customer or a vehicle; not because you need to share a single method.

Maybe we name this something generic, like `LocationAware`?